### PR TITLE
TM: Update title expresssions - variables

### LINKED ISSUE #5753: Expression variables @parent and @layer are poorly documented
## Description

I couldn't find any description of where these variable come from and how they can be used.

@parent is mentioned e.g. in https://docs.qgis.org/testing/en/docs/user_manual/working_with_vector/expression.html#aggregates-functions

@layer can apparently be used in the Field calculator expressions (see attached screenshot) when using the Graphical Modeler

It would be very helpful to have short descriptions how to make use of these features!

qgis_layer_variable

### PR DESCRIPTION
**Pull request by havatv**
Goal: TM: Update title expresssions - variables

Ticket(s): #5753

- [x] Backport to LTR documentation is required

### CODE REVIEW

File: docs/user_manual/working_with_vector/expression.rst [hunk 1]
```diff
@@ -569,7 +569,7 @@ This group contains general assorted functions.
    "is_layer_visible", "Returns true if a specified layer is visible"
    "layer_property", "Returns a property of a layer or a value of its metadata.
    It can be layer name, crs, geometry type, feature count..."
-   "var", "Returns the value stored within a specified variable. See variable functions below"
+   "var", "Returns the value stored within a specified variable. See *variables* below"
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/expression.rst:572]: Better use a ref here instead
```suggestion
   "var", "Returns the value stored within a specified variable. See :ref:`variables_functions`"
```

**Comment by DelazJ** [comment]
Training manual?

File: docs/user_manual/working_with_vector/expression.rst (hunk 1 above)
Comment (havatv) [reply docs/user_manual/working_with_vector/expression.rst:572]: Funny you should suggest this. I have already fixed it (and also change the anchor/label name).

File: docs/user_manual/working_with_vector/expression.rst [hunk 2]
```diff
@@ -569,19 +569,19 @@ This group contains general assorted functions.
...
    where the same value needs to be used more than once."
 
 
 .. _geometry_functions:
 
-Geometry Functions
-------------------
-
-This group contains functions that operate on geometry objects (e.g., length, area).
+Geometry Functions and Variables
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/expression.rst:580]: The group is called "geometry" in the GUI, nothing related to variables.
This makes things confusing IMO

File: docs/user_manual/working_with_vector/expression.rst [hunk 3]
```diff
@@ -569,19 +569,19 @@ This group contains general assorted functions.
...
+--------------------------------
 
+This group contains functions and variables (prefixed with a '$') that
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/expression.rst:583]: I'm not sure the '$' functions should be considered "variables". These are functions that return properties of the features themselves; Not sure they fall in what QGIS calls variables, ie the ones that begin with '@'. And most of the time, if I'm not wrong, a variable can be overridden in a lower level than where it's defined. Not sure these properties can.

File: docs/user_manual/working_with_vector/expression.rst [hunk 4]
```diff
@@ -1051,10 +1051,10 @@ This group contains functions to operate on raster layer.
 
 .. _record_attributes:
 
-Record and Attributes Functions
--------------------------------
+Record and Attributes
+---------------------
 
-This group contains functions that operate on record identifiers.
+This group contains functions and variables (prefixed with a '$') that operate on record identifiers.
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/expression.rst:1057]: ditto

**Comment by DelazJ** [comment]
> Fix #5753

@havatv how does this PR fix the reported issue? 
- `@parent` is already mentioned (despite it's not listed as variable in any dialogs in QGIS afaik)
- `@layer` is also already mentioned

but
- the screenshot in the report shows some input related variables in the modeler context (note that they are different from the `@layer` variable we list) --> we miss modeler context variables (generation)
- the request is also about providing examples of how these variables can be used 

IMO these are what should be addressed before we claim we fix the report.

File: docs/user_manual/working_with_vector/expression.rst (hunk 3 above)
Comment (havatv) [reply docs/user_manual/working_with_vector/expression.rst:583]: OK. I thought that the $ prefix indicated that these were variables (as in PHP). I will fix.

**Pull request merged**